### PR TITLE
make ancient appending default to packing

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Problem
the way of the future is packing ancient append vecs.

#### Summary of Changes
Set the default to packing ancient append vecs. This has been stable for a long time.

Fixes #

### CODE REVIEW

File: accounts-db/src/accounts_db.rs [hunk 1]
```diff
@@ -2513,7 +2513,7 @@ impl AccountsDb {
         let create_ancient_storage = accounts_db_config
             .as_ref()
             .map(|config| config.create_ancient_storage)
-            .unwrap_or(CreateAncientStorage::Append);
+            .unwrap_or(CreateAncientStorage::Pack);
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs]: Can this use the default instead? Then we only need to manage one location.

```suggestion
            .unwrap_or_default());
```

File: accounts-db/src/accounts_db.rs [hunk 2]
```diff
@@ -4400,7 +4400,9 @@ impl AccountsDb {
             .slots_considered
             .fetch_add(1, Ordering::Relaxed);
 
-        if is_ancient(accounts) {
+        // if an append vec is at least 80% of the ideal capacity of an ancient append vec, that's close enough.
+        // If we packed, then we end up allocating exact size ancient append vecs. Those will likely never be exactly the ideal ancient capacity.
+        if accounts.capacity() * 100 / get_ancient_append_vec_capacity() > 80 {
```
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs:4405]: Is this change intended for this PR?

File: accounts-db/src/accounts_db.rs
Comment (brooksprumo) [inline accounts-db/src/accounts_db.rs:1]: Can line 2382 be updated to use the default? Right now it specifies `Pack` explicitly.

current:
```rust
        AccountsDb {
            create_ancient_storage: CreateAncientStorage::Pack,
```

suggested:
```rust
        AccountsDb {
            create_ancient_storage: CreateAncientStorage::default(),
```

File: accounts-db/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:4405]: i just put it in. we could break it out. it fixes a hole in the append code that is really bad.

File: accounts-db/src/accounts_db.rs
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:1]: i'd like it at pack. I don't know that the default matters anyway for accountsdb except for tests?

Comment (brooksprumo) [reply accounts-db/src/accounts_db.rs:1]: This PR sets the default for `CreateAncientStorage` to Pack, so the behavior will be the same. By using `default()`, then we have one fewer place that is special-cased in case we want/need to change the default in the future.

Is there a reason *not* to use `default()` here?

File: accounts-db/src/accounts_db.rs (hunk 2 above)
Comment (brooksprumo) [reply accounts-db/src/accounts_db.rs:4405]: Does this new logic only apply here in `should_move_to_ancient_accounts_file()`, or should it go into `is_ancient()`?

File: accounts-db/src/accounts_db.rs
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:1]: pack is the future. it is the plan. it is the only one I want to have to support.
this one is already at pack. I'd like it to never go backwards?
I'd like to get rid of the whole enum at some point.

Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:1]: not worth arguing over. I changed it.

File: accounts-db/src/accounts_db.rs (hunk 2 above)
Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:4405]: i think the only other call of `is_ancient` is now the code below and that code only counts if we are appending, which will have ideal sized append vecs:
```
    fn is_candidate_for_shrink(
        &self,
        store: &AccountStorageEntry,
        allow_shrink_ancient: bool,
    ) -> bool {
        // appended ancient append vecs should not be shrunk by the normal shrink codepath.
        // It is not possible to identify ancient append vecs when we pack, so no check for ancient when we are not appending.
        let total_bytes = if self.create_ancient_storage == CreateAncientStorage::Append
            && is_ancient(&store.accounts)
```

Comment (jeffwashington) [reply accounts-db/src/accounts_db.rs:4405]: so it is fine to put the code in like this, imo. The hope is the append path is dead soon anyway.

**Review by brooksprumo — APPROVED** [review]
:shipit:
